pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+                cls.responses['publish'],
+            ) for i in range(2)
+        ))
+        for i in range(2):
+            for call_report in cls.bodies['publish'][i]:
+                poll_spawned_tasks(cls.cfg, call_report)
+
+        # Query modules on server. Data integrity is tested in tests below.
+        hrefs = []
+        hrefs.append(''.join((
+            '/v3/releases?module=',
+            _PUPPET_QUERY_PACKAGE['name'],
+            '/',
+            _PUPPET_QUERY_PACKAGE['author']
+        )))
+        hrefs.append(''.join((
+            '/v3/releases?module=',
+            _PUPPET_QUERY_PACKAGE['name'],
+            '/',
+            _PUPPET_QUERY_PACKAGE['author']
+        )))
+        cls.bodies['query units'] = tuple((
+            _query_repo(
+                cls.cfg,
+                hrefs[i],
+                cls.bodies['distribute'][i]['repo_id'],
+                cls.responses['query units']
+            ) for i in range(2)
+        ))
+
+        # Download units from locations returned by query above
+        responses = tuple((
+            requests.get(
+                cls.cfg.base_url +
+                query_body['results'][0]['file_uri'],
+                **cls.cfg.get_requests_kwargs()
+            ) for query_body in cls.bodies['query units']
+        ))
+        cls.downloaded_puppets = tuple((
+            response.content for response in responses
+        ))
+
+        # Search for all units in each of the two repositories.
+        cls.bodies['search units'] = tuple((
+            _get_units(cls.cfg, attrs['_href'], cls.responses['search units'])
+            for attrs in cls.attrs_iter
+        ))
+
+    def test_sanity(self):
+        """Verify we collected a correct set of response bodies.
+
+        This test does not verify Pulp behaviour. Rather, it helps to ensure
+        that this test is written correctly.
+
+        """
+        self.assertEqual(set(self.responses.keys()), set(self.bodies.keys()))
+
+    def test_status_codes(self):
+        """Verify the HTTP status code of each server response."""
+        steps_codes = (
+            ('upload_malloc', 201),
+            ('upload', 200),
+            ('import', 202),
+            ('upload_free', 200),
+            ('copy', 202),
+            ('distribute', 201),
+            ('publish', 202),
+            ('search units', 200),
+            ('query units', 200),
+        )
+        for step, code in steps_codes:
+            with self.subTest((step, code)):
+                for response in self.responses[step]:
+                    self.assertEqual(response.status_code, code)
+
+    def test_call_reports_keys(self):
+        """Verify each call report has the correct keys."""
+        for step in {'import', 'copy'}:
+            with self.subTest(step=step):
+                self.assertEqual(
+                    set(self.bodies[step].keys()),
+                    {'error', 'result', 'spawned_tasks'},
```

Comment:
Looks like [1] it's fixed. So I would make another PR for it.

[1] http://pulp.readthedocs.org/en/latest/dev-guide/conventions/sync-v-async.html#call-report